derive_builder_core/src/builder.rs

```diff
@@ -140,6 +140,48 @@ pub struct Builder<'a> {
     pub std: bool,
 }
 
+impl<'a> Builder<'a> {
+    #[doc = "
+        adding builder() fn on the original struct
+        ```rust
+        impl <original_struct> {
+         pub fn builder() -> <original_struct>Builder  {
+             <original_struct>Builder::default()
+         }
+        }
+        ```
+        Why ?
+        Because sometimes @alexzanderr likes to call the builder like this:
+        ```rust
+            let ch = Channel::builder()
+            .special_info(42u8)
+            .token(19_124)
+            .build()
+            .unwrap();
+        ```
+        Note the `Channel::builder()`, it generates the builder from the original struct.
+    "]
+    pub fn add_builder_fn_to_struct(&self, ast: &syn::DeriveInput) -> TokenStream {
+        // name of the original struct
+        let struct_name_ident = &ast.ident;
+        // name of the builder for the original struct
+        let builder_name_ident = &self.ident;
+
+        // rust source code
+        let expanded = quote! {
+            /// This returns a new builder for the original struct
+            impl #struct_name_ident {
+                pub fn builder() -> #builder_name_ident {
+                    #builder_name_ident::default()
+                }
+            }
+        };
+
+        // this token stream is appended in in `builder_for_struct`
```

Comment:
you are saying that rust-analyzer will add those comments alongside your actual references objects when running `find all refs` command inside your code editor/IDE.

so for the user developer those comments will be garbadge data? right?

question: is that why in all serious rust crates I've never seen a single comment? and also thats why your crate doesnt have any comments?

if so: but man, we need to document our code, how do you know what it does if you come over the code after 1 year without scratching your head off?

Reply:
I can't speak to why every crate behaves a certain way, and there are plenty of examples of comments in this crate ([example](https://github.com/colin-kiegel/rust-derive-builder/blob/master/derive_builder_core/src/initializer.rs)).

My tendency is to document the _outside_ of something if - when I come back to the code after some time away - I find myself confused about what the body of that thing is doing. That approach tends to have two positive effects:
1. I know which pieces of the code need further explanation; this can be hard to determine immediately after writing the code
2. The comment is unlikely to become stale without the person changing the file seeing the comment. Stale comments are worse than no comments because they can cause people to question the comments they do see in the codebase.